cli/cmd/common/transaction.go

```diff
@@ -92,7 +92,7 @@ func SignConsensusTransaction(
 		}
 
 		// Gas estimation if not specified.
-		if tx.Fee.Gas == invalidGasLimit {
+		if tx.Fee.Gas == invalidGasLimit && wallet.ConsensusSigner() != nil {
```

Comment:
This check should probably be at the top and it should immediately return an error (e.g. the check from lines 121-124 should be moved to the top).